tests/Shared/Aspire.Workload.Testing.targets

```diff
@@ -76,18 +76,69 @@
 
+  <UsingTask TaskName="PrepareNuGetConfigForWorkloadTesting"
+      TaskFactory="RoslynCodeTaskFactory"
+      AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.Core.dll">
+    <ParameterGroup>
+      <TemplateNuGetConfigPath ParameterType="System.String" Required="true" />
+      <OutputFile ParameterType="System.String" Required="true" />
+      <AddBuiltNuGetSource ParameterType="System.Boolean" Required="false" />
+      <AddPackageSourceMapping ParameterType="System.Boolean" Required="false" />
+    </ParameterGroup>
+    <Task>
+      <Using Namespace="System.IO" />
+      <Using Namespace="System.Xml.Linq" />
+      <Using Namespace="System.Xml.XPath" />
+      <Code Type="Fragment" Language="cs">
+        <![CDATA[
+          if (!File.Exists(TemplateNuGetConfigPath))
+          {
+            Log.LogError($"Could not find nuget config template at '{TemplateNuGetConfigPath}' .");
+            return false;
+          }
+
+          XDocument doc = XDocument.Load(TemplateNuGetConfigPath);
+          if (AddBuiltNuGetSource)
+          {
+            string xpath = "/configuration/packageSources";
+            XElement packageSources = doc.XPathSelectElement(xpath);
+            if (packageSources is null)
+            {
+              Log.LogError($"Could not find {xpath} in {TemplateNuGetConfigPath}");
+              return false;
+            }
+            packageSources.LastNode.AddAfterSelf(
+              new XElement("add",
+                new XAttribute("key", "nuget-local"),
+                new XAttribute("value", "%BUILT_NUGETS_PATH%")));
+          }
+
+          if (AddPackageSourceMapping)
+          {
+            string mappingXpath = "/configuration/packageSourceMapping";
+            XElement packageSourceMapping = doc.XPathSelectElement(mappingXpath);
+            if (packageSourceMapping is null)
+            {
+              // if mapping has been removed completely then the task needs an update!
+              throw new InvalidOperationException($"Expected to find {mappingXpath} in {TemplateNuGetConfigPath}");
+            }
+
+            packageSourceMapping.FirstNode.AddBeforeSelf(
+              new XElement("packageSource",
+                new XAttribute("key", "nuget-local"),
+                new XElement("package", new XAttribute("pattern", "*Aspire*"))));
+          }
+          doc.Save(OutputFile);
+          Log.LogMessage(MessageImportance.Low, $"Generated patched nuget.config at {OutputFile}");
+        ]]>
+      </Code>
+    </Task>
+  </UsingTask>
+
   <!-- For test projects -->
 
-  <Target Name="_PatchNuGetConfigForBuildingTestProject"
-          AfterTargets="GetCopyToOutputDirectoryItems"
-          Condition="'$(InstallWorkloadForTesting)' == 'true'"
-          Inputs="$(TemplateNuGetConfigPath)"
-          Outputs="$(PatchedNuGetConfigPath)">
-    <!-- %BUILT_NUGETS_PATH% is set when building the testproject -->
-    <PatchNuGetConfig TemplateNuGetConfigPath="$(RepoRoot)NuGet.config"
```

Comment:
We could still use this task (and not have to make our own RoslynCodeTaskFactory task) if we switch around how this Target gets pulled in, right?

For example, if we conditionally add our target to `$(GetCopyToOutputDirectoryItemsDependsOn)` based on `$(InstallWorkloadForTesting)`, then the target won't get pulled in at all unless that bool is set.

Reply:
This target would be needed from VS when running tests-outside-of-repo. That's why we need a net472 task assembly too.
I will see if I can make the task net472 compatible, but for now this will fix the immediate issue and not regress.

Reply:
> I will see if I can make the task net472 compatible

This sounds like a good option. Thanks.